.automation/copy-issue-labels-to-pr

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/python3
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import re
+import sys
+from github import Github
+
+# Get inputs from shell
+(token, repository, path) = sys.argv[1:4]
+
+# We don't want to copy all labels on linked issues; only those in this subset.
+COPYABLE_LABELS = ["amd sev","documentation","ibm pef","infrastructure","intel sgx","wasm"]
+
+# Authenticate with Github using our token
+g = Github(token)
+
+# Initialize repo
+repo = g.get_repo(repository)
+
+# Open Github event JSON
+with open(path) as f:
+    event = json.load(f)
+
+# --- Extract PR from event JSON ---
+# `check_run`/`check_suite` does not include any direct reference to the PR
+# that triggered it in its event JSON. We have to extrapolate it using the head
+# SHA that *is* there.
+
+# Get head SHA from event JSON
+pr_head_sha = event["check_suite"]["head_sha"]
+
+# Get all repo pull requests
+repo_pull_requests = repo.get_pulls()
+
+# Find the repo PR that matches the head SHA we found
+pr = {pr.head.sha: pr for pr in repo_pull_requests}[pr_head_sha]
+
+# Extract all associated issues from closing keyword in PR
+regex = re.compile("(close[sd]?|fix|fixe[sd]?|resolve[sd]?)\s*:?\s+#(\d+)", re.I)
+closing_numbers = [number for keyword, number in regex.findall(pr.body)]
+if len(closing_numbers) == 0:
+    quit()
+
+# Get the superset of every label on every linked issue, filtered by our 
+# acceptable labels list.
+labels_to_add = []
+for number in closing_numbers:
+    for label in repo.get_issue(int(number)).labels:
+        if label not in labels_to_add and label.name in COPYABLE_LABELS:
+            labels_to_add += [label]
```

Comment:
This section is screaming for a set comprehension and intersection rather than list searches.